lib/yarp.rb

```diff
@@ -3,19 +3,19 @@
 module YARP
   # This represents a location in the source corresponding to a node or token.
   class Location
-    attr_reader :start_offset, :end_offset
+    attr_reader :start_offset, :length
 
-    def initialize(start_offset, end_offset)
+    def initialize(start_offset, length)
       @start_offset = start_offset
-      @end_offset = end_offset
+      @length = length
     end
 
-    def length
-      @end_offset - @start_offset
+    def end_offset
+      @start_offset + @length
     end
 
     def deconstruct_keys(keys)
-      { start_offset: start_offset, end_offset: end_offset }
+      { start_offset: start_offset, end_offset: end_offset } # TODO
```

Comment:
I was wondering if we should change the `deconstruct_keys` to include `length` instead of `end_offset`.
But it doesn't seem so useful and things like `pretty_print` should still show the end_offset as that seems more readable, and would be noisy to change.
I'll just remove it, I can try it later and see whether that makes sense.